DigitalLearningSolutions.Data/Services/DiagnosticAssessmentDataService.cs

```diff
@@ -29,6 +29,61 @@ public DiagnosticAssessmentDataService(IDbConnection connection, ILogger<Diagnos
             DiagnosticAssessment? diagnosticAssessment = null;
             return connection.Query<DiagnosticAssessment, DiagnosticTutorial, DiagnosticAssessment>(
                 @"
+                    WITH NextTutorialAndSectionNumbers AS (
+                        SELECT TOP (1)
+                            MIN(NextTutorials.TutorialID) AS NextTutorialId,
+                            MIN(NextSections.SectionID) AS NextSectionId
+                        FROM Sections
+                            INNER JOIN Customisations
+                                ON Customisations.ApplicationID = Sections.ApplicationID
+                            INNER JOIN Applications
+                                ON Applications.ApplicationID = Sections.ApplicationID
+                                
+                            LEFT JOIN CustomisationTutorials AS NextSectionCustomisationTutorials
+                                ON NextSectionCustomisationTutorials.CustomisationID = Customisations.CustomisationID
+                            LEFT JOIN Tutorials AS NextSectionTutorials
+                                ON NextSectionTutorials.TutorialID = NextSectionCustomisationTutorials.TutorialID
+                                AND NextSectionTutorials.ArchivedDate IS NULL
+                            LEFT JOIN Sections AS NextSections
+                                ON NextSections.SectionID = NextSectionTutorials.SectionID
+                                AND NextSections.ArchivedDate IS NULL
+                                AND (
+                                    Sections.SectionNumber < NextSections.SectionNumber
+                                    OR Sections.SectionID < NextSections.SectionID
+                                )
```

Comment:
We also need to check that the next section number is `<=` the current section, else this will give us any sections with a smaller section number if they happen to have a larger section ID. In no case should we get a next section that has a smaller section number to the current one, it should either have a larger section number (and any ID it likes) or have the same section number, and a larger ID 🙂